TextToSpeech/TextToSpeechJsonRpc.cpp

```diff
@@ -95,20 +97,21 @@ namespace Plugin {
             TTSLOG_INFO("SetACL request:%s\n",parameters["accesslist"].String().c_str());
             JsonArray list = parameters["accesslist"].Array();
             JsonArray::Iterator it = list.Elements();
+            uint32_t status;
             while(it.Next())
             {
                 JsonObject accesslist = it.Current().Object();
                 if (accesslist.HasLabel("method") && accesslist.HasLabel("apps"))
                 {
-                    _tts->SetACL(accesslist["method"].String(),accesslist["apps"].String());
+                    status = _tts->SetACL(accesslist["method"].String(),accesslist["apps"].String());
```

Comment:
How is the above comment handled?

Reply:
if(method != "speak") {
           return Core::ERROR_GENERAL;
        }

        if(apps.empty() || apps.c_str()[0] == ' ' || apps == "NULL") {
           return Core::ERROR_GENERAL;
        }
        
        Have added this checks in the SetACL method.

Reply:
No I believe, the setACL is implemented in a way that it can accept access control directives for more than on methods. For eg 
[
    {"method": "speak", "apps": ["RA", "SAD", "HtmlApp-0"]}
    {"method": "enabletts", "apps" : ["RA"]},
    {"method": "setconfiguration", "apps" : ["RA", "SAD"]}
]

In that example, if the second item (or even the first for that matter) has any issue / error, it will be shadowed by the next item's status. So, you should bail out from the while loop as soon as you detect the first error.

Reply:
Now, I am breaking the while loop as soon as the status == Core::ERROR_GENERAL.